web/src/main/java/com/graphhopper/http/SimpleRouteSerializer.java

```diff
@@ -100,6 +101,16 @@ private String getMessage(Throwable t) {
 
                     jsonPath.put("legs", ar.getLegs());
 
+                    List<PathDetails> details = ar.getPathDetails();
+                    if (!details.isEmpty()) {
+                        //TODO I don't really like the conversion, should we keep it like that in the PathWrapper?
+                        Map<String, Map<Object, List<int[]>>> pathDetailsMap = new HashMap<>();
+                        for (PathDetails pathDetails : details) {
+                            pathDetailsMap.put(pathDetails.getName(), pathDetails.getPathDetailsMap());
+                        }
+                        jsonPath.put("details", pathDetailsMap);
```

Comment:
Can't we use a Map when returning from ar.getPathDetails so that we could write jsonPath.put("details", ar.getPathDetails())? Or annotate PathDetails so that we get the JSON output that we want via jackson.